### ISSUE TITLE
Elasticsearch startup fails with 'failed to obtain data locks'

### ISSUE THREAD

**Issue by dustinchilson** [upmc-enterprises/elasticsearch-operator#117]
I've been experimenting with the azure example and am getting the above issue.

Could be related to pires/kubernetes-elasticsearch-cluster#145

```YAML
apiVersion: enterprises.upmc.com/v1
kind: ElasticsearchCluster
metadata:
  name: graylog-es
spec:
  client-node-replicas: 1
  master-node-replicas: 3
  data-node-replicas: 6
  network-host: 0.0.0.0
  zones: []
  data-volume-size: 10Gi
  java-options: "-Xms256m -Xmx256m"
  snapshot:
    scheduler-enabled: false
    bucket-name: elasticsnapshots99
    cron-schedule: "@every 2m"
  storage:
    storage-class: default
  resources:
    requests:
      memory: 512Mi
      cpu: 500m
    limits:
      memory: 512Mi
      cpu: '1'
```

Thanks for any direction you might be able to give me.

**Comment by stevesloka** [comment]
Hey @dustinchilson, this is similar to #116. Just curious what version of k8s are you running?

**Comment by dustinchilson** [comment]
I was attempting on v1.8.2. I built my cluster using [Azure/acs-engine](https://github.com/Azure/acs-engine) from source.

I have 3 Standard_D2_v2 VMs (2 Core, 7gb ram) for nodes.

I can try a 1.7 cluster if that helps.

**Comment by dustinchilson** [comment]
I tried on a v1.7.9 cluster and received the same messages.

```
[2017-11-07T14:33:46,911][INFO ][o.e.n.Node               ] [6b9a5f86-238f-4479-b25c-276e74d9a66e] initializing ...
[2017-11-07T14:33:47,002][WARN ][o.e.b.ElasticsearchUncaughtExceptionHandler] [6b9a5f86-238f-4479-b25c-276e74d9a66e] uncaught exception in thread [main]
org.elasticsearch.bootstrap.StartupException: java.lang.IllegalStateException: failed to obtain node locks, tried [[/data/data/myesdb]] with lock id [0]; maybe these locations are not writable or multiple nodes were started without increasing [node.max_local_storage_nodes] (was [1])?
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:127) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:114) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:58) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:122) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.cli.Command.main(Command.java:88) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:91) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:84) ~[elasticsearch-5.3.1.jar:5.3.1]
Caused by: java.lang.IllegalStateException: failed to obtain node locks, tried [[/data/data/myesdb]] with lock id [0]; maybe these locations are not writable or multiple nodes were started without increasing [node.max_local_storage_nodes] (was [1])?
	at org.elasticsearch.env.NodeEnvironment.<init>(NodeEnvironment.java:260) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.node.Node.<init>(Node.java:262) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.node.Node.<init>(Node.java:242) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.bootstrap.Bootstrap$6.<init>(Bootstrap.java:242) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:360) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:123) ~[elasticsearch-5.3.1.jar:5.3.1]
	... 6 more
Caused by: java.io.IOException: failed to obtain lock on /data/data/nodes/0
	at org.elasticsearch.env.NodeEnvironment.<init>(NodeEnvironment.java:239) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.node.Node.<init>(Node.java:262) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.node.Node.<init>(Node.java:242) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.bootstrap.Bootstrap$6.<init>(Bootstrap.java:242) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:360) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:123) ~[elasticsearch-5.3.1.jar:5.3.1]
	... 6 more
Caused by: java.io.IOException: Mount point not found
	at sun.nio.fs.LinuxFileStore.findMountEntry(LinuxFileStore.java:91) ~[?:?]
	at sun.nio.fs.UnixFileStore.<init>(UnixFileStore.java:65) ~[?:?]
	at sun.nio.fs.LinuxFileStore.<init>(LinuxFileStore.java:44) ~[?:?]
	at sun.nio.fs.LinuxFileSystemProvider.getFileStore(LinuxFileSystemProvider.java:51) ~[?:?]
	at sun.nio.fs.LinuxFileSystemProvider.getFileStore(LinuxFileSystemProvider.java:39) ~[?:?]
	at sun.nio.fs.UnixFileSystemProvider.getFileStore(UnixFileSystemProvider.java:368) ~[?:?]
	at java.nio.file.Files.getFileStore(Files.java:1461) ~[?:1.8.0_121]
	at org.elasticsearch.env.ESFileStore.getMatchingFileStore(ESFileStore.java:107) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.env.Environment.getFileStore(Environment.java:351) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.env.NodeEnvironment$NodePath.<init>(NodeEnvironment.java:108) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.env.NodeEnvironment.<init>(NodeEnvironment.java:227) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.node.Node.<init>(Node.java:262) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.node.Node.<init>(Node.java:242) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.bootstrap.Bootstrap$6.<init>(Bootstrap.java:242) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:360) ~[elasticsearch-5.3.1.jar:5.3.1]
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:123) ~[elasticsearch-5.3.1.jar:5.3.1]
```

**Comment by stevesloka** [comment]
I think a newer elastic image should work. I'm working on that right now. Will update very soon!

**Comment by dustinchilson** [comment]
Just checking in, Has there been any progress on this. 

Thanks

**Comment by stevesloka** [comment]
Yup I've got good progress. Using the newer elastic image fixes, I've got a big update to support 1.8 clusters since annotations moved.

**Comment by stevesloka** [comment]
@dustinchilson I think I've got this fixed up in #123 If you want to test sooner here are the relevant images:

- upmcenterprises/elasticsearch-operator:0.0.10-beta
- upmcenterprises/docker-elasticsearch-kubernetes:5.6.4_1

Also, I've had issues running in minikube, but this works fine in a regular cluster (AWS or Bare Metal)
